pyop2/base.py

```diff
@@ -998,6 +1065,12 @@ def __str__(self):
     def __repr__(self):
         return "MixedSet(%r)" % (self._sets,)
 
+    def __eq__(self, other):
+        try:
+            return self._sets == other._sets
+        except AttributeError:
```

Comment:
Couldn't you write `return type(self) == type(other) and self._sets == other._sets` instead of exception handling?